aspnetcore/mvc/views/partial.md

```diff
@@ -1,173 +1,277 @@
+* Reduce the duplication of common markup content across markup files.
 
-ASP.NET Core MVC controllers have a [PartialView](/dotnet/api/microsoft.aspnetcore.mvc.controller.partialview#Microsoft_AspNetCore_Mvc_Controller_PartialView) method which is called from an action method. Razor Pages have no equivalent `PartialView` method on the [PageModel](/dotnet/api/microsoft.aspnetcore.mvc.razorpages.pagemodel).
+  When the same markup elements are used across markup files, a partial view removes the duplication of markup content into one partial view file. When the markup is changed in the partial view, it updates the rendered output of the markup files that use the partial view.
+
+Partial views shouldn't be used to maintain common layout elements. Common layout elements should be specified in [_Layout.cshtml](xref:mvc/views/layout) files.
+
+Don't use a partial view where complex rendering logic or code execution is required to render the markup. Instead of a partial view, use a [view component](xref:mvc/views/view-components).
 
 ## Declare partial views
 
-Partial views are created like a regular view&mdash;by creating a *.cshtml* file within the *Views* folder. There's no semantic difference between a partial view and a regular view; however, they're rendered differently. You can have a view that's returned directly from a controller's [ViewResult](/dotnet/api/microsoft.aspnetcore.mvc.viewresult), and the same view can be used as a partial view. The main difference between how a view and a partial view are rendered is that partial views don't run *_ViewStart.cshtml*. Regular views do run *_ViewStart.cshtml*. Learn more about *_ViewStart.cshtml* in [Layout](xref:mvc/views/layout)).
+::: moniker range=">= aspnetcore-2.1"
+
+A partial view is a *.cshtml* markup file maintained within the *Views* folder (MVC) or *Pages* folder (Razor Pages).
+
+In ASP.NET Core MVC, a controller's <xref:Microsoft.AspNetCore.Mvc.ViewResult> is capable of returning either a view or a partial view. An analogous capability is planned for Razor Pages in ASP.NET Core 2.2. In Razor Pages, a <xref:Microsoft.AspNetCore.Mvc.RazorPages.PageModel>'s <xref:Microsoft.AspNetCore.Mvc.IActionResult> can return a <xref:Microsoft.AspNetCore.Mvc.PartialViewResult>. Referencing and rendering partial views is described in detail in the next section.
 
-As a convention, partial view file names often begin with `_`. This naming convention isn't a requirement, but it helps to visually differentiate partial views from regular views.
+Unlike view or page rendering, a partial view doesn't run *_ViewStart.cshtml*. For more information on *_ViewStart.cshtml*, see <xref:mvc/views/layout>.
+
+Partial view file names often begin with an underscore (`_`). This naming convention isn't required, but it helps to visually differentiate partial views from views and pages. Razor Pages ignores files that start with an underscore, even if the page includes an `@page` directive.
```

Comment:
Clarify that you're talking about MVC in the first sentence.

Is there a way to override this restriction? I can imagine someone, including myself, wanting to avoid renaming partial views just because I'm using Razor Pages.

Reply:
> Is there a way to override this restriction? I can imagine someone, including myself, wanting to avoid renaming partial views just because I'm using Razor Pages.

I don't follow u there. Why would you need to rename partial views with RP?

Reply:
I may have misunderstood the last sentence in that paragraph. I took it to mean that my partial view file name must not begin with underscore if I want to return it from a Razor Pages page model method.

Reply:
Yeah ... I see what you mean now. I took "ignores" to just mean that the partial view won't be processed as a Razor Pages page (even if it has the directive), so I need to rephrase it to say exactly that. I'll fix it on the next commit. Thanks for pointing it out. I'm sure it would have been a problem for readers in this format.